src/main/java/teammates/common/datatransfer/questions/FeedbackRubricQuestionDetails.java

```diff
@@ -196,13 +272,31 @@ private boolean isValidDescriptionSize() {
         return true;
     }
 
+    /**
+     * Checks if the dimensions of rubricWeightsForEachCell is valid according
+     * to numOfRubricSubQuestions and numOfRubricChoices.
+     */
+    private boolean isValidWeightSize() {
+        if (rubricWeightsForEachCell.size() != numOfRubricSubQuestions) {
+            return false;
+        }
+        for (int i = 0; i < rubricWeightsForEachCell.size(); i++) {
+            if (rubricWeightsForEachCell.get(i).size() != numOfRubricChoices) {
```

Comment:
Would `stream` and `allMatch` make the code clearer here? :D

Reply:
I am not very familiar with `stream`, I am studying it while I am working on unit tests. I will try to make the code cleaner like you suggested within the next few pushes.